bridge-impl/src/main/java/com/liferay/faces/bridge/event/internal/HeaderRequestPhaseListener.java

```diff
@@ -38,31 +38,30 @@
 	// serialVersionUID
 	private static final long serialVersionUID = 8470095938465172618L;
 
-	// Protected (Lazy-Initialized) Constants
-	private static Boolean VIEW_PARAMETERS_ENABLED;
-
-	// Private Data Members
-	private PhaseId phaseId = PhaseId.RESTORE_VIEW;
+	// Instance field must be declared volatile in order for the double-check idiom to work (requires JRE 1.5+)
+	private static volatile Boolean viewParametersEnabled;
 
 	@Override
 	public void afterPhase(PhaseEvent phaseEvent) {
 
 		FacesContext facesContext = phaseEvent.getFacesContext();
 
-		if (VIEW_PARAMETERS_ENABLED == null) {
+		// First check without locking (not yet thread-safe)
+		if (viewParametersEnabled == null) {
 
 			synchronized (this) {
```

Comment:
@ngriffin7a, thanks so much for finding this! Even though the classes you mentioned may work correctly with `synchronized (this)` (since they are likely application scoped singletons), I decided to synchronize on the `class` instead. I think that's more appropriate since we're dealing with `static` variables here (and who knows who might copy this code?). I have sent three PRs to fix this:

https://github.com/liferay/liferay-faces-bridge-impl/pull/285
https://github.com/liferay/liferay-faces-bridge-impl/pull/286
https://github.com/liferay/liferay-faces-showcase/pull/87